### PR TITLE
Implement download()

### PR DESCRIPTION
**Pull request by JohanLorenzo**
A lot of copy and paste. I tested the download and checksums locally.

### CODE REVIEW

File: staging_upload.py [hunk 1]
```diff
@@ -17,9 +21,14 @@
 max_concurrent_aiohttp_streams = 10
 # map to copy artifacts to bucket - public/build/target.complete.mar
 
-# where are the mac / win mars?
-# https://archive.mozilla.org/pub/devedition/tinderbox-builds/jamun-macosx64-devedition/1494964280/
 
+LATEST_MAR_LOCATIONS = {
+    'win32': 'https://index.taskcluster.net/v1/task/gecko.v2.jamun.latest.devedition.win32-opt/artifacts/public/build/firefox-54.0.en-US.win32.complete.mar',
+    'win64': 'https://index.taskcluster.net/v1/task/gecko.v2.jamun.latest.devedition.win64-opt/artifacts/public/build/firefox-54.0.en-US.win64.complete.mar',
```
Comment (escapewindow) [inline staging_upload.py:27]: I'd love to see these templatized, so someone could specify 'jamun' or 'mozilla-beta', and the version number, and get the expected mar locations.  (My intent was to take the hardcoded vars at the top, and turn them into commandline options via argparse at some point.)

File: staging_upload.py [hunk 2]
```diff
@@ -36,9 +45,133 @@ def die(msg):
...
-async def download():
-    pass
-    # windows / mac download from archive; linux download from task
+async def download(context):
+    abs_path_per_platform = {}
+    for platform, mar_url in LATEST_MAR_LOCATIONS.items():
+        platform_sub_folder = get_platform_base_folder(context, platform)
+
+        mar_abs_path = os.path.join(platform_sub_folder, get_sub_path_from_url(mar_url))
+        await download_file(context, url=mar_url, abs_filename=mar_abs_path)
```
Comment (escapewindow) [inline staging_upload.py:54]: This flow will result in synchronous downloading, which is probably fine with 5 mars to download.
If we want to speed things up, we can do something like

```python
tasks = []
tasks.append(
    asyncio.ensure_future(
        download_file(context, url=mar_url, abs_filename=mar_abs_path)
    )
)
await raise_future_exceptions(tasks)
```

where `raise_future_exceptions` is https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/utils.py#L320-L327

**Review by escapewindow — APPROVED** [review]
Lgtm!  Let's merge, and we can add `upload` and any further improvements later.  Also, I'm expecting the final file to land somewhere else (tools? braindump?) though keeping staging-release-upload as a github repo under mozilla-releng might be ok too.

File: staging_upload.py (hunk 2 above)
Comment (JohanLorenzo) [reply staging_upload.py:54]: Thanks for pointing it out. I wasn't sure where I got the logic wrong :)

**Pull request merged**